tensorboard/plugins/pr_curve/summary.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from google.protobuf import json_format
+from tensorboard.plugins.pr_curve import pr_curve_pb2
+
+def op(
+    tag,
+    labels,
+    predictions,
+    num_thresholds=200,
+    weights=None,
+    num_classes=1,
+    display_name=None):
+  """Computes multi-class PR summaries for a list of thresholds in `[0, 1]`.
+
+  Computes true/false positive/negative values for the given `predictions`
+  against the ground truth `labels`, against a list of evenly distributed
+  threshold values in `[0, 1]` of length `num_thresholds`.
+
+  Each number in `predictions`, a float in `[0, 1]`, is compared with its
+  corresponding label in `labels`, and counts as a single tp/fp/tn/fn value at
+  each threshold. This is then multiplied with `weights` which can be used to
+  reweight certain values, or more commonly used for masking values.
+
+  This method supports multi-class classification, and when `num_classes > 1`,
+  the last dimension of `labels` and `predictions` is the class dimension.
+
+  Like most `tf.metrics` methods, this returns a pair `(metrics, update_op)`
+  where `update_op` should be used at each `session.run()` to update `metrics`,
+  and `metrics` should be evaluated at the end.
+
+  NOTE(chizeng): This is basically a faster implementation of simliar methods in
+  `tf.contrib.metrics.streaming_XXX_at_thresholds`, where we assume the
+  threshold values are evenly distributed and thereby can implement a `O(n+m)`
+  algorithm instead of `O(n*m)` in both time and space, where `n` is the
+  size of `labels` and `m` is the number of thresholds.
+
+  Args:
+    tag: A tag attached to the summary. Used by TensorBoard for organization.
+    labels: The ground truth values, a `Tensor` whose dimensions must match
+        `predictions`. Should be of type `bool`.
+    predictions: A floating point `Tensor` whose values are in the range
+        `[0, 1]`. Shape is arbitrary if `num_classes=1`; otherwise, the
+        last dimension should be exactly `num_classes`.
+    num_thresholds: Number of thresholds, evenly distributed in `[0, 1]`, to
```

Comment:
Docs for `num_thresholds` should mention that this must be a constant `int` and not an integer `Tensor`.